packages/paste-core/components/avatar/__tests__/avatar.test.tsx

```diff
@@ -52,36 +54,44 @@ describe('Avatar', () => {
       });
     });
 
+    describe('getCorrespondingIconSizeFromSizeToken', () => {
+      it('should return a reduced sizeIcon to match icon size', () => {
+        expect(getCorrespondingIconSizeFromSizeToken('sizeIcon10')).toEqual('sizeIcon10');
+        expect(getCorrespondingIconSizeFromSizeToken('sizeIcon20')).toEqual('sizeIcon10');
+        expect(getCorrespondingIconSizeFromSizeToken('sizeIcon30')).toEqual('sizeIcon10');
+        expect(getCorrespondingIconSizeFromSizeToken('sizeIcon40')).toEqual('sizeIcon10');
+        expect(getCorrespondingIconSizeFromSizeToken('sizeIcon50')).toEqual('sizeIcon20');
+        expect(getCorrespondingIconSizeFromSizeToken('sizeIcon60')).toEqual('sizeIcon20');
+        expect(getCorrespondingIconSizeFromSizeToken('sizeIcon70')).toEqual('sizeIcon30');
+        expect(getCorrespondingIconSizeFromSizeToken('sizeIcon80')).toEqual('sizeIcon40');
+        expect(getCorrespondingIconSizeFromSizeToken('sizeIcon90')).toEqual('sizeIcon50');
+        expect(getCorrespondingIconSizeFromSizeToken('sizeIcon100')).toEqual('sizeIcon70');
+        expect(getCorrespondingIconSizeFromSizeToken('sizeIcon110')).toEqual('sizeIcon80');
```

Comment:
These tests are for the utils function that gets its argument from the Avatar props. Do you mean we should test if Avatar is given a size value that isn't an iconSize type? Would typing cover that possibility and prevent it from getting to a helper function like this?

Reply:
Typescript typings provide safety for teams that develop using Typescript, but not for teams using Javascript. It is also handy to test error cases, because in the future if this function is refactored (say in 2 years by someone else), they can use the tests to understand the behavior better and avoid introducing a regression.

Reply:
To check for passing in "wrong" values, we changed the utils function to check for size being string and including "sizeIcon" in it. We created a new test to check that "wrong" values throw an error.